binderhub/health.py

```diff
@@ -155,58 +127,43 @@ async def check_docker_registry(self):
+    def get_checks(self, checks):
+        """Add health checks to the `checks` dict
 
-        quota = self.settings["pod_quota"]
-        total_pods = n_user_pods + n_build_pods
-        usage = {
-            "total_pods": total_pods,
-            "build_pods": n_build_pods,
-            "user_pods": n_user_pods,
-            "quota": quota,
-            "ok": total_pods <= quota if quota is not None else True,
-        }
-        return usage
+        checks: Dictionary, updated in-place:
+          key: service name
+          value: a future that resolves to either:
+            - a bool (success/fail)
+            - a dict with the field `"ok": bool` plus other information
+        """
+        if self.settings["use_registry"]:
+            checks["Docker registry"] = self.check_docker_registry()
+        checks["JupyterHub API"] = self.check_jupyterhub_api(self.hub_url)
 
     async def check_all(self):
-        """Runs all health checks and returns a tuple (overall, checks).
+        """Runs all health checks and returns a tuple (overall, results).
 
         `overall` is a bool representing the overall status of the service
-        `checks` contains detailed information on each check's result
+        `results` contains detailed information on each check's result
         """
-        checks = []
-        check_futures = []
-
-        if self.settings["use_registry"]:
-            check_futures.append(self.check_docker_registry())
-            checks.append({"service": "Docker registry", "ok": False})
-
-        check_futures.append(self.check_jupyterhub_api(self.hub_url))
-        checks.append({"service": "JupyterHub API", "ok": False})
-
-        check_futures.append(self.check_pod_quota())
-        checks.append({"service": "Pod quota", "ok": False})
+        checks = {}
+        results = []
+        self.get_checks(checks)
 
-        for result, check in zip(await asyncio.gather(*check_futures), checks):
+        for result, service in zip(
+            await asyncio.gather(*checks.values()), checks.keys()
+        ):
             if isinstance(result, bool):
-                check["ok"] = result
+                results.append({"service": service, "ok": result})
             else:
-                check.update(result)
+                results.append(dict({"service": service}, **result))
 
-        # The pod quota is treated as a soft quota this means being above
-        # quota doesn't mean the service is unhealthy
-        overall = all(
-            check["ok"] for check in checks if check["service"] != "Pod quota"
-        )
+        # Some checks are for information but do not count as a health failure
+        overall = all(r["ok"] for r in results if not r.get("_ignored", False))
         if not overall:
-            unhealthy = [check for check in checks if not check["ok"]]
+            unhealthy = [r for r in results if not r["ok"]]
             app_log.warning(f"Unhealthy services: {unhealthy}")
```

Comment:
Here we will warn about the `_ignored` services as well and not hint that they are accepted to fail. But hmmm, we have the `_ignored` field as part of the `r` dictionary, so maybe its fine as the reader could deduce that.

I think about the naming `_ignore` though, maybe we can make it more descriptive? Its a "check" in a "health check" context, and we don't consider the health to be bad / not ok if this check is failing. What about `_accept_failure` or something like that? We can go with `_ignore` as well if this doesn't resonate with you!

Reply:
```python
# Some checks are for information but do not count as a health failure
```

Maybe even `_count_as_health_failure`?